server/src/catalog.rs

```diff
@@ -216,72 +311,248 @@ fn read_model_table_metadata(
 
         let mut arrow_reader = ParquetFileArrowReader::new(Arc::new(reader));
         let mut record_batch_reader = arrow_reader.get_record_reader(row_count)?;
-        let rows = record_batch_reader.next().unwrap()?; //TODO: handle empty Parquet files
-
-        let sampling_intervals = extract_and_shift_int32_column(&rows, 2)?;
-        let denormalized_dimensions = extract_and_shift_denormalized_dimensions(&rows, 4)?;
-
-        Ok(Arc::new(ModelTableMetadata {
-            name: table_name,
-            segment_folder: table_folder + "/segment",
-            segment_group_file_schema: segment_group_file_schema.clone(),
-            sampling_intervals,
-            denormalized_dimensions,
-        }))
-    } else {
-        Err(ParquetError::General(format!(
-            "unable to read metadata for {}",
-            table_name
-        )))
+        let rows = record_batch_reader.next().ok_or_else(|| {
+            ParquetError::General(format!("No metadata exists for {}.", *table_name))
+        })??;
+        Ok(rows)
     }
-}
 
-fn extract_and_shift_int32_column(
-    rows: &RecordBatch,
-    column_index: usize,
-) -> Result<Int32Array, ParquetError> {
-    let column = rows
-        .column(column_index)
-        .as_any()
-        .downcast_ref::<Int32Array>()
-        .unwrap()
-        .values();
-    let mut shifted_column = Int32Builder::new(column.len() + 1);
-    shifted_column.append_value(-1)?; //-1 is stored at index 0 as ids starting at 1 is used for lookup
-    shifted_column.append_slice(column)?;
-    Ok(shifted_column.finish())
-}
+    /// Check that the model table only use supported model types.
+    fn check_model_type_metdata(table_folder: &String) -> Result<(), ParquetError> {
+        let model_types_file = table_folder.clone() + "/model_type.parquet";
+        let path = Path::new(&model_types_file);
+        if let Ok(file) = File::open(&path) {
+            let reader = SerializedFileReader::new(file)?;
+            let rows = reader.get_row_iter(None)?;
+            for row in rows {
+                let mtid = row.get_int(0)?;
+                let name = row.get_bytes(1)?.as_utf8()?;
+                match (mtid, name) {
+                    (1, "dk.aau.modelardb.core.models.UncompressedModelType") => (),
+                    (2, "dk.aau.modelardb.core.models.PMC_MeanModelType") => (),
+                    (3, "dk.aau.modelardb.core.models.SwingFilterModelType") => (),
+                    (4, "dk.aau.modelardb.core.models.FacebookGorillaModelType") => (),
+                    _ => return Err(ParquetError::General("unsupported model type".to_string())),
```

Comment:
This error message should have punctuation and use `to_owned()`.

Reply:
Fixed.